src/main/java/de/qabel/desktop/DesktopClient.java

```diff
@@ -21,110 +21,150 @@
-		customProperties.put("accountRepository", accountRepository);
-		ClientConfiguration clientConfig = getClientConfiguration(persistence, identityRepository, accountRepository);
-		customProperties.put("clientConfiguration", clientConfig);
-
-		Injector.setConfigurationSource(customProperties::get);
-		Injector.setInstanceSupplier(new RecursiveInjectionInstanceSupplier(customProperties));
-		return clientConfig;
-	}
-
-	private ClientConfiguration getClientConfiguration(Persistence<String> persistence, IdentityRepository identityRepository, AccountRepository accountRepository) {
-		ClientConfigurationRepository repo = new PersistenceClientConfigurationRepository(
-				persistence,
-				new ClientConfigurationFactory(),
-				identityRepository,
-				accountRepository
-		);
-		final ClientConfiguration config = repo.load();
-		config.addObserver((o, arg) -> {
-			repo.save(config);
-		});
-		return config;
-	}
-
-	private void setTrayIcon(Stage primaryStage) {
-
-		SystemTray sTray = SystemTray.getSystemTray();
-
-		ActionListener listenerShow = e -> Platform.runLater(() -> primaryStage.show());
-		ActionListener listenerClose = e -> System.exit(0);
-		primaryStage.setOnCloseRequest(arg0 -> primaryStage.hide());
-
-		PopupMenu popup = new PopupMenu();
-		MenuItem showItem = new MenuItem("Öffnen");
-		MenuItem exitItem = new MenuItem("Beenden");
-
-		showItem.addActionListener(listenerShow);
-		exitItem.addActionListener(listenerClose);
-
-		popup.add(showItem);
-		popup.add(exitItem);
-
-		URL url = System.class.getResource("/logo.png");
-		Image img = Toolkit.getDefaultToolkit().getImage(url);
-		TrayIcon icon = new TrayIcon(img, "Qabel", popup);
-		icon.setImageAutoSize(true);
-
-		try {
-			sTray.add(icon);
-		} catch (AWTException e) {
-			System.err.println(e);
-		}
-
-	}
-
-	public static Account getBoxAccount() {
-		return new Account("http://localhost:9696", "testuser", "testuser");
-	}
+    private static final String TITLE = "Qabel Desktop Client";
+    Scene scene;
+
+    public static void main(String[] args) throws Exception {
+        launch(args);
+    }
+
+    @Override
+    public void start(Stage primaryStage) throws Exception {
+        setUserAgentStylesheet(STYLESHEET_MODENA);
+
+        final Map<String, Object> customProperties = new HashMap<>();
+        ClientConfiguration config = initDiContainer(customProperties);
+
+        if (!config.hasAccount()) {
+            scene = new Scene(new LoginView().getView(), 370, 530);
+            config.addObserver((o, arg) -> {
+                if (arg instanceof Account) {
+                    Scene layoutScene = new Scene(new LayoutView().getView(), 800, 600);
+                    Platform.runLater(() -> primaryStage.setScene(layoutScene));
+                }
+            });
+        } else {
+            scene = new Scene(new LayoutView().getView(), 800, 600);
+        }
+
+        Platform.setImplicitExit(false);
```

Comment:
I think you need a shutdown handler with `Platform.exit()` to shutdown properly.